nornir_netbox/plugins/inventory/netbox.py

```diff
@@ -157,31 +158,59 @@ def __init__(
         self.nb_url = nb_url
         self.flatten_custom_fields = flatten_custom_fields
         self.filter_parameters = filter_parameters
+        self.include_vms = include_vms
 
         self.session = requests.Session()
         self.session.headers.update({"Authorization": f"Token {nb_token}"})
         self.session.verify = ssl_verify
 
     def load(self) -> Inventory:
 
-        url = f"{self.nb_url}/api/dcim/devices/?limit=0"
-        nb_devices: List[Dict[str, Any]] = []
+        nb_devices: List[Dict[str, Any]] = self._get_resource(
+            url=f"{self.nb_url}/api/dcim/devices/?limit=0",
+            params=self.filter_parameters,
+        )
+
+        hosts: Hosts = self._get_hosts(nb_devices)
+
+        if self.include_vms:
+            nb_vms: List[Dict[str, Any]] = self._get_resource(
+                url=f"{self.nb_url}/api/virtualization/virtual-machines/?limit=0",
+                params=self.filter_parameters,
+            )
+
+            hosts.update(self._get_hosts(nb_vms, virtual=True))
+
+        return Inventory(hosts=hosts)
+
+    def _get_resource(self, url: str, params: Dict[str, Any]) -> List[Dict[str, Any]]:
+
+        resource: List[Dict[str, Any]] = []
 
         while url:
-            r = self.session.get(url, params=self.filter_parameters)
+            r = self.session.get(url, params=params)
 
             if not r.status_code == 200:
+                print(r)
                 raise ValueError(
-                    f"Failed to get devices from NetBox instance {self.nb_url}"
+                    f"Failed to get data from NetBox instance {self.nb_url}"
                 )
 
             resp = r.json()
-            nb_devices.extend(resp.get("results"))
+            resource.extend(resp.get("results"))
 
             url = resp.get("next")
 
+        return resource
+
+    def _get_hosts(
```

Comment:
I'm not really sure why you are introducing this function. In my opinion it's not really needed.